src/applications/personalization/profile/components/notification-settings/NotificationRadioButtons.jsx

```diff
@@ -0,0 +1,266 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import _ from 'lodash';
+import classNames from 'classnames';
+import { makeField } from '~/platform/forms/fields';
+
+/**
+ * A radio button group with a label.
+ *
+ * Validation has the following props.
+ *
+ * - `additionalFieldsetClass` String for any additional fieldset classes.
+ * - `additionalLegendClass` String for any additional legend classes.
+ * - `errorMessage` String Error message for the radio button group
+ * - `warningMessage` String Warning message for the radio button group
+ * - `successMessage` String Success message for the radio button group
+ * - `label` String for the group field label.
+ * - `name` String for the name attribute.
+ * - `options` Array of options to populate group.
+ * - `required` is this field required.
+ * - `value` string. Value of the select field.
+ * - `onValueChange` a function with this prototype: (newValue)
+ * - `onMouseDown` a function for mouse down
+ * - `onKeyDown` a function for key down
+ * - `ariaDescribedby` Array matching the options, added when the option is
+ *   selected
+ */
+const NotificationRadioButtons = ({
+  id = _.uniqueId('notification-radio-buttons-'),
+  additionalFieldsetClass,
+  additionalLegendClass,
+  errorMessage,
+  warningMessage,
+  successMessage,
+  label,
+  name,
+  options,
+  required,
+  value,
+  onValueChange,
+  ariaDescribedby,
+  onMouseDown,
+  onKeyDown,
+}) => {
+  const handleChange = domEvent => {
+    const optionValue = domEvent.target.value;
+    // removed analytics here
+    onValueChange(makeField(optionValue, true));
+  };
+
+  let errorSpan = '';
+  let errorSpanId = undefined;
+  if (errorMessage) {
+    errorSpanId = `${id}-error-message`;
+    errorSpan = (
+      <span className="rb-input-error-message" role="alert" id={errorSpanId}>
+        <i className="fas fa-exclamation-circle vads-u-margin-x--1" />{' '}
+        <span className="sr-only">Error</span> {errorMessage}
+      </span>
+    );
+  }
+
+  let warningSpan = '';
+  let warningSpanId = undefined;
+  if (warningMessage) {
+    warningSpanId = `${id}-warning-message`;
+    warningSpan = (
+      <span
+        className="rb-input-warning-message"
+        role="alert"

```

Comment:
we might need to remove this alert role from the warning as elements with alert roles can be overly assertive and annoying. but let's leave it for now.

Reply:
np good to know. Wasn't sure if that would be a thing axe would complain about